src/main/java/touro/chess/UndoManager.java

```diff
@@ -0,0 +1,89 @@
+package touro.chess;
+
+import java.util.Stack;
+/*
+    TO DO:
+    clean code
+    fix naming?
+    private class within public?
+    testing issues
+    other scenarios? if a piece is changed? don't know chess well enough...
+
+ */
+public class UndoManager { //don't like naming
```

Comment:
That's mainly what I wanted to discuss with you.

Reply:
I'm trying to figure out how to verify that the MoveInfo object is created properly (contains the right move and captured piece) but because MoveInfo is an inner private class, I cannot create a MoveInfo object in my tests